server/pbenchinacan/pod-nodata.yml

```diff
@@ -70,6 +77,21 @@ spec:
       - name: vm.max_map_count
         value: "262144"
     workingDir: /usr/share/elasticsearch
+  - command:
+    - /opt/keycloak/bin/kc.sh
+    - start-dev
+    env:
+    - name: KEYCLOAK_ADMIN
+      value: admin
```

Comment:
Your PR description said `test_admin`; but I like this better, since I think we want to always have a real `admin` user with a well-documented mechanism to change the default password (which might be just editing a copy of this pod YML).

Reply:
I don't see any point in a `test_admin` user *in addition to* the `admin` user. I definitely want to have `admin` available for functional testing, although we're going to have to work out the mechanics of Keycloak authentication for the functional tests. (E.g., if we want to use username/password then we need to do this through Keycloak APIs; whereas when we work out an API Token mechanism we'll need to figure out how to generate one of those and make it available to the functional test system as a secret.)

Reply:
The `admin` user referenced here is for the master realm; the `test_admin` would be a user in the Pbench realm.

Reply:
OK; the context isn't that obvious here. Still, let's not have a `test_admin`, regardless of the realm. It serves no purpose when we need a real `admin` anyway.

Reply:
No, the context was easy to miss.  It wasn't until after Nikhil and I talked (after he confirmed that the settings here don't work anyway), that we realized the misapprehension.

Reply:
just an FYI: There is a difference between, the `master` realm `admin` user vs the `test_admin` user that I have added in the `pbench` realm. The `test_admin` user does not have access like the `admin` user does because the `test_admin` does not have the role `admin` from the master realm. However, I have also created a custom `ADMIN` role under the `pbench-client` (like we have right now in our user table source code) that can be assigned to users (that we create during the functional test) if needs to be. 

When we log in `test_admin` (or any user with `ADMIN` role) the login token will include the custom role `ADMIN` that we can check after decoding the token and take action accordingly. This, therefore, does not need a real `admin` user for doing privileged things on the Pbench server. We can create as many roles as we want later under our `pbench-client` client. 

The reason I added `ADMIN` role to the image is that then we can reuse the `is_admin` functionality that we have by just verifying the login token and check if the role is present.